### ISSUE TITLE
Avoid virtual columns for secondary index MVs

### ISSUE THREAD

**Issue by duarten** [scylladb/scylla#3833]
Secondary indexes maintain a materialized view that maps an indexed base table column to its key, which is then used to query the base table. As #2994 points out, this allows us to solve inconsistencies that cause the view to contain ghost entries.

In materialized views, virtual columns ensure that an entry that was added because of an unselected column being alive (e.g., `update cf set unselected_col = x where k = y`) are kept alive for as long as that unselected column is live too. They apply only to views that do not include a base table regular column in their primary key.

We can take advantage of index verification repair to remove outdated view entries, and we can insert never-expiring rows into the view (non-ttl'd live row marker). This allows us to generate very few materialized view updates for these secondary indexes.

**Comment by nyh** [comment]
@duarten looks good. I wonder if it should be optional - could we have a use case where the user relies on rows being added and disappearing all the time and will suffer from a huge number of zombie rows remaining in the view and needing to skipped during reads?

By the way, please note that as test_3362_no_ttls() shows, before we added virtual cells we had a problem not **just** with TTLs. It was also possible to add and delete two different columns with different timestamps (see details in that test) in a way which made it impossible to create the row marker for the empty view row with any timestamp. So we may need to not just ignore TTLs, but also unselected column deletions. I think that's fine - I think we don't need any further changes to ignore those deletions, after we dropped the complex (and unfounded) row marker timestamp calculation. But we'll need to consider this again - I'm not sure if the current row marker timestamp calculation is appropriate here.

**Comment by duarten** [comment]
Yes, we can make it optional (or default to true and have a knob). We could also have the manager monitor such secondary indexes, and once in a while do a table scan.
